src/main/kotlin/org/amshove/kluent/BasicBacktick.kt

```diff
@@ -2,18 +2,20 @@ package org.amshove.kluent
 
 import kotlin.reflect.KClass
 
-infix fun Any?.`should equal`(theOther: Any?) = this.shouldEqual(theOther)
+infix fun <T> T.`should equal`(theOther: T?) : T = this.shouldEqual(theOther)
 
 infix fun Any?.`should not equal`(theOther: Any?) = this.shouldNotEqual(theOther)
```

Comment:
These could also be chained with generics. 
`"some String".shouldNotEqual("other").shouldNotBeBlank()`